### ISSUE TITLE
Status bar is hidden aperiodically, after MPGNotification is shown

### ISSUE THREAD

**Issue by jeanclad** [MPGNotification/MPGNotification#26]
Status bar is hidden aperiodically, after MPGNotification is shown.

**Comment by toblerpwn** [comment]
I don't have an easy project to test this on, so hopefully someone else can pick this up in the mean time.

**Comment by toblerpwn** [comment]
also @jeanclad - can you confirm which version of the repo you are using? are you using Cocoapods? if so, which version?

there were significant changes in version 1.2, so I'm wondering if you were using version 1.1 or earlier.

**Comment by hagile** [comment]
This is still happening, I'm using latest classes from git only. When I show around 10 notifications and interact with any of the pop up, it'll hide the status bar. @jeanclad  @toblerpwn

**Comment by mzekrallah** [comment]
yes, the status bar hides and shows every 5 or 6 subsequent enqueues of MPGNotifications.

**Comment by ahmedjafri** [comment]
I'm still seeing this with version 1.2 on Cocoapods.

**Comment by toblerpwn** [comment]
Can you confirm that it is the latest, v1.2.4?

Also can you provide sample code for how you call it? Specifically I am wondering whether you are adding it to the `mainWindow` or a specific `hostViewController`. In my personal projects, I have migrated away from the 'window' approach, and instead always add it to a view controller using either the `MPGNotification notificationWithHostViewController: ...` initializer, or setting `notification.hostViewController` prior to calling any of the `show ...` method.

**Comment by ahmedjafri** [comment]
`MPGNotification *notification = [MPGNotification notificationWithTitle:@"New Message"
                                                                  subtitle:@"test message"
                                                           backgroundColor:[Theme getMainColor]
                                                                 iconImage:nil];`
    `notification.duration = 5.0f;`
    `[notification show];`

We're calling it without specifying a `hostViewController`. In our app, `show` is called in AppDelegate when we recieve a remote notification at which point in time we don't know the currently showing UIViewController. I suppose we could add some logic to maintain that state and call show with a `hostViewController`.

**Comment by taher-mosbah** [comment]
@ahmedjafri @toblerpwn any update on this please ? I'm having the same problem.

**Comment by fmessina** [comment]
@taher-mosbah @ahmedjafri @toblerpwn the problem is due to this
`self.windowLevel = [[[[UIApplication sharedApplication] delegate] window] windowLevel];`
in the creation of the notification.

I solved this issue by sharing a single notification object among the entire app (in a singleton), then I make sure if I want to present a second notification that the first one is dismissed.

`        MPGNotification *currentNotification = [SharedComponents sharedInstance].currentMessageNotification;
        if (currentNotification) {
            [currentNotification dismissWithAnimation:NO];
        }
`
